libdino/src/service/connection_manager.vala

```diff
@@ -243,19 +243,15 @@ public class ConnectionManager {
     }
 
     private bool network_is_online() {
-        if (network_monitor != null && network_monitor.network_available && network_monitor.connectivity == GLib.NetworkConnectivity.FULL) {
-            return true;
-        }
-
-        return false;
+        return network_monitor != null && network_monitor.network_available;
```

Comment:
As I said in a private chat, I'd like to note here (so that it's written down somewhere public) that this isn't right and the connectivity check really is the correct thing to do. Also, to make sure everyone in the Dino team understands the situation.

* `network_available` means that your computer has network routes to "somewhere". That is, it is connected to a router.
* `connectivity.FULL` means that your computer can access "the internet". That is, if you're behind a router, that router is connected.

As a result, only checking for `network_available` is not correct.

Unfortunately, NetworkManager tends to wait a long time before checking for connectivity. As a result, it is possible that a transient network error leaves NetworkManager thinking that you have `network_available=true` but `connectivity!=FULL`, and it will wait several minutes before realizing that your Internet connexion did come back.

During that time, apps checking for connectivity (e.g the whole GNOME desktop) will think they don't have access to the Internet, while apps that don't (e.g Firefox) will access the Internet just fine. Users are understandably confused when that happens.

And this is what @fiaxh is worried about and wants to avoid.

Removing the check for connectivity might be an acceptable trade-off in the short-term (that's really not for me to decide, it's the job of a maintainer), but it seems to me like we should go and get NetworkManager fixed, so that it avoids these situations. I had told @fiaxh I would at least go and talk to the NetworkManager folks, so I guess I need to own up to that. 🙂

Reply:
Here it is: https://bugzilla.gnome.org/show_bug.cgi?id=792240

I added a comment in the code and the commit message linking to this bug report, so we eventually add the connectivity check back when NetworkManager is improved.